### PR TITLE
Further external C cleanup, added external/signal.ooc

### PR DESCRIPTION
**Pull request by marcusnaslund**
Part of the work on #1484

### CODE REVIEW

File: source/system/external/windows.ooc [hunk 1]
```diff
@@ -7,7 +7,7 @@
  */
 
 version(windows) {
-include windows | (_WIN32_WINNT=0x0500)
+include windows | (NOMINMAX, _WIN32_WINNT=0x0500)
```
Comment (marcusnaslund) [inline source/system/external/windows.ooc]: http://stackoverflow.com/questions/14657474/conflict-of-windows-h-and-limits

File: source/system/Numbers.ooc [hunk 2]
```diff
@@ -6,10 +6,7 @@
  * of the MIT license.  See the LICENSE file for details.
  */
 
-include stdlib, stdint, stddef, float, ctype, sys/types
-version(windows || apple) {
-	include limits
-}
```
Comment (marcusnaslund) [inline source/system/Numbers.ooc]: Was included for windows and apple here, and included for linux previously in `Exception.ooc`.

**Comment by marcusnaslund** [comment]
Peer review mr. Win.. uh, ehm, @sebastianbaginski ?

File: source/system/external/windows.ooc (hunk 1 above)
Comment (sebastianbaginski) [inline source/system/external/windows.ooc]: I remember a funny compilation error related to some windows headers being included in wrong order, so yes, this does not surprise me...

**Pull request merged**